File: 백준/Silver/10828. 스택/스택.py
#10828
import sys

class Stack:
    def __init__(self):
        self.stack = []

    def push(self, num):
        self.stack.append(num)

    def pop(self):
        if self.empty():
            return -1
        return self.stack.pop()

    def top(self):
        if self.empty():
            return -1
        return self.stack[-1]

    def size(self):
        return len(self.stack)

    def empty(self):
        return int(self.size() == 0)


stack = Stack()
n = int(sys.stdin.readline())

for _ in range(n):
    command = sys.stdin.readline().split()

    if command[0] == 'push':
        stack.push(int(command[1]))
    elif command[0] == 'pop':
        print(stack.pop())
    elif command[0] == 'top':
        print(stack.top())
    elif command[0] == 'size':
        print(stack.size())
    elif command[0] == 'empty':
        print(stack.empty())

File: 백준/Silver/10828. 스택/스택.cc
//10828

#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {
	int n, num;
	int result = 0;
	cin >> n;

	stack<int> s;
	string cmd;

	for (int i = 0; i < n; ++i) {
		cin >> cmd;

		if (cmd == "push") {
			cin >> num;
			s.push(num);
		}

		else if (cmd == "pop") {
			if (s.size() == 0) {
				result = -1;
				cout << result << endl;
			}
			else {
				result = s.top();
				cout << result << endl;
				s.pop();
			}
		}

		else if (cmd == "size") {
			cout << s.size() << endl;
		}

		else if (cmd == "empty") {
			if (s.size() == 0) {
				result = 1;
				cout << result << endl;
			}
			else {
				result = 0;
				cout << result << endl;
			}
		}

		else if (cmd == "top") {
			if (s.size() == 0) {
				result = -1;
				cout << result << endl;
			}
			else {
				result = s.top();
				cout << result << endl;
			}
		}
	}
}